### PR TITLE
reporting unused imports

### LINKED ISSUE #18: Implement detection and reporting of unused imports
https://github.com/Microsoft/vscode-python/issues/43
https://github.com/Microsoft/PTVS/issues/254

### PR DESCRIPTION
**Pull request by heejaechang**
Fixes #18.

made basic reporting working.

- [x] basic reporting
- [x] use correct message on diagnostic
- [x] Add support for the new unnecessary tag to have fading effect (https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/#diagnostic)
- [x] check VS and vscode whether it supports new unnecessary tag spec
- [x] add tests

... I am splitting out code fix part to different PR
- [ ] add remove unused import code fix
- [ ] etc (might have more work item once I get to remove unused import such as grouping unused imports based on what gets removed together on 1 code fix and etc)
...

this is with latest lsp client ("vscode-languageclient@6.0.0-next.1") + latest vscode.
without it installed, you will get "..." rather than fading out.

unusedimport

...

to get new lsp client that supports fading, do this.

first, take changes in https://github.com/heejaechang/vscode-python/tree/newlspclient
and then do npm install and npm ci

### CODE REVIEW

File: src/Analysis/Ast/Impl/Linting/UnusedImports/UnusedImportsLinter.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            var variableDeclared = analysis.GlobalScope.Variables;
+            foreach (var name in imported.Names) {
+                if (!imported.TryGetVariable(name, out var variableFromImportCollection)) {
+                    continue;
+                }
+
+                // we have variable from import statement, but we don't have any variable declared from actual
+                // usage. meaning the import is not used.
+                if (!variableDeclared.TryGetVariable(name, out var variableFromVariableCollection)) {
+                    AppendDiagnostic(variableFromImportCollection, ref result);
```
Comment (heejaechang) [inline src/Analysis/Ast/Impl/Linting/UnusedImports/UnusedImportsLinter.cs]: this check might be all I need if what it does is by design.

**Comment by heejaechang** [comment]
@Anapo14 can you take a look to see whether this meets your expectations?

@MikhailArkhipov @jakebailey @bschnurr can you code review? this is reporting part. code fix part is blocked by my "add import" PR since it require code action portion I added there.

File: src/Analysis/Ast/Impl/Linting/UnusedImports/UnusedImportsLinter.cs (hunk 1 above)
Comment (heejaechang) [reply src/Analysis/Ast/Impl/Linting/UnusedImports/UnusedImportsLinter.cs]: not true for cases where we could delete duplicated imports for same symbol.

**Comment by Anapo14** [comment]
> @Anapo14 can you take a look to see whether this meets your expectations?

based on the gif, it looks good to me!

**Pull request closed without merging**